src/codegen/sys/lib_.rs

```diff
@@ -265,29 +244,11 @@ fn generate_constants(w: &mut Write, env: &Env, constants: &[Constant]) -> Resul
             }
         }
 
-        if env.library
-            .type_(constant.typ)
-            .maybe_ref_as::<Bitfield>()
-            .is_some()
-        {
-            try!(writeln!(
-                w,
-                "{}pub const {}: {} = {2} {{ bits: {} }};",
-                comment,
-                constant.c_identifier,
-                type_,
-                value
-            ));
-        } else {
-            try!(writeln!(
-                w,
-                "{}pub const {}: {} = {};",
-                comment,
-                constant.c_identifier,
-                type_,
-                value
-            ));
-        }
+        try!(writeln!(
+            w,
+            "{}pub const {}: {} = {};",
+            comment, constant.c_identifier, type_, value
+        ));
```

Comment:
There will be problem with this for negative const's like `G_LOG_LEVEL_MASK`,
so maybe better restore special case for `Bitfield` with same action as in bitfield generation?

Reply:
ack

Reply:
This was actually never handled with casting, and G_LOG_LEVEL_MASK is not
generated by this code but rather as a part of generate_bitfields, so there is
no pressing need to fix this as a part of this PR, but in general it would be
nice for consistency.